server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -335,9 +331,8 @@ protected static void parseProperties(ObjectMapper.Builder objBuilder, Map<Strin
         } else {
             this.mappers = CopyOnWriteHashMap.copyOf(mappers);
         }
-        this.nestedTypePathAsString = "__" + fullPath;
-        this.nestedTypePathAsBytes = new BytesRef(nestedTypePathAsString);
-        this.nestedTypeFilter = new TermQuery(new Term(TypeFieldMapper.NAME, nestedTypePathAsBytes));
+        this.nestedTypePath = "__" + fullPath;
```

Comment:
For indices created in v8 we don't need to add any prefix since we use a dedicated field ?

Reply:
++ I pushed 7ea0da8